packages/forklift-console-plugin/src/modules/Mappings/mappingActions.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useMemo } from 'react';
+import { ConfirmModal } from 'legacy/src/common/components/ConfirmModal';
+import { AddEditMappingModal } from 'legacy/src/Mappings/components/AddEditMappingModal';
+import { useDeleteMappingMutation } from 'legacy/src/queries';
+import { Mapping, MappingType } from 'legacy/src/queries/types';
+import { useTranslation } from 'src/utils/i18n';
+
+import { withActionContext } from '@kubev2v/common/components/ActionServiceDropdown';
+import withQueryClient from '@kubev2v/common/components/QueryClientHoc';
+import { useModal } from '@kubev2v/common/polyfills/sdk-shim';
+
+import { CommonMapping } from './dataCommon';
+import { FlatNetworkMapping } from './dataForNetwork';
+import { FlatStorageMapping } from './dataForStorage';
+
+export const useNetworkMappingActions = ({ entity }: { entity: FlatNetworkMapping }) =>
+  useMappingActions<FlatNetworkMapping>({ entity, mappingType: MappingType.Network });
+
+export const useStorageMappingActions = ({ entity }: { entity: FlatStorageMapping }) =>
+  useMappingActions<FlatStorageMapping>({ entity, mappingType: MappingType.Storage });
+
+function useMappingActions<T extends CommonMapping>({
+  entity,
+  mappingType,
+}: {
+  entity: T;
+  mappingType: MappingType;
+}) {
+  const { t } = useTranslation();
+  const launchModal = useModal();
+  const areProvidersReady = entity.sourceReady && entity.targetReady;
+  const areProvidersResolved = entity.sourceResolved && entity.targetResolved;
+  const editingDisabled = !areProvidersReady || !areProvidersResolved;
+
+  const disabledTooltip = !areProvidersResolved
+    ? t(
+        'This mapping cannot be edited because it includes missing source or target resources. Delete and recreate the mapping.',
+      )
+    : t(
+        'This mapping cannot be edited because the inventory data for its associated providers is not ready',
+      );
+
+  const actions = useMemo(
+    () => [
+      {
+        id: 'edit',
+        cta: () =>
+          launchModal(withQueryClient(EditModal), {
+            mapping: entity.object,
+            mappingType,
+            namespace: entity.namespace,
+          }),
+        label: t('Edit Mapping'),
+        disabled: editingDisabled,
+        disabledTooltip: editingDisabled ? disabledTooltip : '',
+      },
+      {
+        id: 'delete',
+        cta: () =>
+          launchModal(withQueryClient(DeleteModal), {
+            mapping: entity.object,
+            mappingType,
+            namespace: entity.namespace,
+            name: entity.name,
+          }),
+        label: t('Delete Mapping'),
+      },
+    ],
+    [t, editingDisabled, disabledTooltip, entity],
+  );
+
+  return [actions, true, undefined];
+}
+
+const EditModal = ({
```

Comment:
```suggestion
const EditMappingModal = ({
```

we may want a general purpose `EditModal` ?